File: OTCDerivativesCalculatorModule/Project_CSharp/XmlBuilder/XmlSerializedClass/fpml-ird-5-4/StubCalculationPeriodAmount.cs
using System;
using System.Text;
using System.Collections.Generic;
using System.Xml;

namespace FpmlSerializedCSharp
{
    public class StubCalculationPeriodAmount : ISerialized
    {
        public StubCalculationPeriodAmount(XmlNode xmlNode)
        {
            XmlNode calculationPeriodDatesReferenceNode = xmlNode.SelectSingleNode("calculationPeriodDatesReference");
            
            if (calculationPeriodDatesReferenceNode != null)
            {
                if (calculationPeriodDatesReferenceNode.Attributes["href"] != null || calculationPeriodDatesReferenceNode.Attributes["id"] != null) 
                {
                    if (calculationPeriodDatesReferenceNode.Attributes["id"] != null) 
                    {
                        calculationPeriodDatesReferenceIDRef_ = calculationPeriodDatesReferenceNode.Attributes["id"].Value;
                        CalculationPeriodDatesReference ob = new CalculationPeriodDatesReference(calculationPeriodDatesReferenceNode);
                        IDManager.SetID(calculationPeriodDatesReferenceIDRef_, ob);
                    }
                    else if (calculationPeriodDatesReferenceNode.Attributes["href"] != null)
                    {
                        calculationPeriodDatesReferenceIDRef_ = calculationPeriodDatesReferenceNode.Attributes["href"].Value;
                    }
                    else
                    {
                        calculationPeriodDatesReference_ = new CalculationPeriodDatesReference(calculationPeriodDatesReferenceNode);
                    }
                }
                else
                {
                    calculationPeriodDatesReference_ = new CalculationPeriodDatesReference(calculationPeriodDatesReferenceNode);
                }
            }
            
        
            XmlNode initialStubNode = xmlNode.SelectSingleNode("initialStub");
            
            if (initialStubNode != null)
            {
                if (initialStubNode.Attributes["href"] != null || initialStubNode.Attributes["id"] != null) 
                {
                    if (initialStubNode.Attributes["id"] != null) 
                    {
                        initialStubIDRef_ = initialStubNode.Attributes["id"].Value;
                        StubValue ob = new StubValue(initialStubNode);
                        IDManager.SetID(initialStubIDRef_, ob);
                    }
                    else if (initialStubNode.Attributes["href"] != null)
                    {
                        initialStubIDRef_ = initialStubNode.Attributes["href"].Value;
                    }
                    else
                    {
                        initialStub_ = new StubValue(initialStubNode);
                    }
                }
                else
                {
                    initialStub_ = new StubValue(initialStubNode);
                }
            }
            
        
            XmlNode finalStubNode = xmlNode.SelectSingleNode("finalStub");
            
            if (finalStubNode != null)
            {
                if (finalStubNode.Attributes["href"] != null || finalStubNode.Attributes["id"] != null) 
                {
                    if (finalStubNode.Attributes["id"] != null) 
                    {
                        finalStubIDRef_ = finalStubNode.Attributes["id"].Value;
                        StubValue ob = new StubValue(finalStubNode);
                        IDManager.SetID(finalStubIDRef_, ob);
                    }
                    else if (finalStubNode.Attributes["href"] != null)
                    {
                        finalStubIDRef_ = finalStubNode.Attributes["href"].Value;
                    }
                    else
                    {
                        finalStub_ = new StubValue(finalStubNode);
                    }
                }
                else
                {
                    finalStub_ = new StubValue(finalStubNode);
                }
            }
            
        
        }
        
    
        #region CalculationPeriodDatesReference_
        private CalculationPeriodDatesReference calculationPeriodDatesReference_;
        public CalculationPeriodDatesReference CalculationPeriodDatesReference_
        {
            get
            {
                if (this.calculationPeriodDatesReference_ != null)
                {
                    return this.calculationPeriodDatesReference_; 
                }
                else if (this.calculationPeriodDatesReferenceIDRef_ != null)
                {
                    calculationPeriodDatesReference_ = IDManager.getID(calculationPeriodDatesReferenceIDRef_) as CalculationPeriodDatesReference;
                    return this.calculationPeriodDatesReference_; 
                }
                else
                {
                      return this.calculationPeriodDatesReference_; 
                }
            }
            set
            {
                if (this.calculationPeriodDatesReference_ != value)
                {
                    this.calculationPeriodDatesReference_ = value;
                }
            }
        }
        #endregion
        
        public string calculationPeriodDatesReferenceIDRef_ { get; set; }
        #region InitialStub_
        private StubValue initialStub_;
        public StubValue InitialStub_
        {
            get
            {
                if (this.initialStub_ != null)
                {
                    return this.initialStub_; 
                }
                else if (this.initialStubIDRef_ != null)
                {
                    initialStub_ = IDManager.getID(initialStubIDRef_) as StubValue;
                    return this.initialStub_; 
                }
                else
                {
                      return this.initialStub_; 
                }
            }
            set
            {
                if (this.initialStub_ != value)
                {
                    this.initialStub_ = value;
                }
            }
        }
        #endregion
        
        public string initialStubIDRef_ { get; set; }
        #region FinalStub_
        private StubValue finalStub_;
        public StubValue FinalStub_
        {
            get
            {
                if (this.finalStub_ != null)
                {
                    return this.finalStub_; 
                }
                else if (this.finalStubIDRef_ != null)
                {
                    finalStub_ = IDManager.getID(finalStubIDRef_) as StubValue;
                    return this.finalStub_; 
                }
                else
                {
                      return this.finalStub_; 
                }
            }
            set
            {
                if (this.finalStub_ != value)
                {
                    this.finalStub_ = value;
                }
            }
        }
        #endregion
        
        public string finalStubIDRef_ { get; set; }
        public string choiceStr_0;
        
    
        
    
    }
    
}

File: OTCDerivativesCalculatorModule/Project_Cpp/lib_static/FpmlSerialized/GenClass/fpml-ird-5-4/StubCalculationPeriodAmount.cpp
// StubCalculationPeriodAmount.cpp 
#include "StubCalculationPeriodAmount.hpp"
#ifdef ConsolePrint
    #include <iostream>
#endif 

namespace FpmlSerialized {

StubCalculationPeriodAmount::StubCalculationPeriodAmount(TiXmlNode* xmlNode)
: ISerialized(xmlNode)
{
    #ifdef ConsolePrint
        std::string initialtap_ = FileManager::instance().tap_;
        FileManager::instance().tap_.append("   ");
    #endif 
   //calculationPeriodDatesReferenceNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* calculationPeriodDatesReferenceNode = xmlNode->FirstChildElement("calculationPeriodDatesReference");

   if(calculationPeriodDatesReferenceNode){calculationPeriodDatesReferenceIsNull_ = false;}
   else{calculationPeriodDatesReferenceIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- calculationPeriodDatesReferenceNode , address : " << calculationPeriodDatesReferenceNode << std::endl;
   #endif
   if(calculationPeriodDatesReferenceNode)
   {
      if(calculationPeriodDatesReferenceNode->Attribute("href") || calculationPeriodDatesReferenceNode->Attribute("id"))
      {
          if(calculationPeriodDatesReferenceNode->Attribute("id"))
          {
             calculationPeriodDatesReferenceIDRef_ = calculationPeriodDatesReferenceNode->Attribute("id");
             calculationPeriodDatesReference_ = boost::shared_ptr<CalculationPeriodDatesReference>(new CalculationPeriodDatesReference(calculationPeriodDatesReferenceNode));
             calculationPeriodDatesReference_->setID(calculationPeriodDatesReferenceIDRef_);
             IDManager::instance().setID(calculationPeriodDatesReferenceIDRef_,calculationPeriodDatesReference_);
          }
          else if(calculationPeriodDatesReferenceNode->Attribute("href")) { calculationPeriodDatesReferenceIDRef_ = calculationPeriodDatesReferenceNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { calculationPeriodDatesReference_ = boost::shared_ptr<CalculationPeriodDatesReference>(new CalculationPeriodDatesReference(calculationPeriodDatesReferenceNode));}
   }

   //initialStubNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* initialStubNode = xmlNode->FirstChildElement("initialStub");

   if(initialStubNode){initialStubIsNull_ = false;}
   else{initialStubIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- initialStubNode , address : " << initialStubNode << std::endl;
   #endif
   if(initialStubNode)
   {
      if(initialStubNode->Attribute("href") || initialStubNode->Attribute("id"))
      {
          if(initialStubNode->Attribute("id"))
          {
             initialStubIDRef_ = initialStubNode->Attribute("id");
             initialStub_ = boost::shared_ptr<StubValue>(new StubValue(initialStubNode));
             initialStub_->setID(initialStubIDRef_);
             IDManager::instance().setID(initialStubIDRef_,initialStub_);
          }
          else if(initialStubNode->Attribute("href")) { initialStubIDRef_ = initialStubNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { initialStub_ = boost::shared_ptr<StubValue>(new StubValue(initialStubNode));}
   }

   //finalStubNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* finalStubNode = xmlNode->FirstChildElement("finalStub");

   if(finalStubNode){finalStubIsNull_ = false;}
   else{finalStubIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- finalStubNode , address : " << finalStubNode << std::endl;
   #endif
   if(finalStubNode)
   {
      if(finalStubNode->Attribute("href") || finalStubNode->Attribute("id"))
      {
          if(finalStubNode->Attribute("id"))
          {
             finalStubIDRef_ = finalStubNode->Attribute("id");
             finalStub_ = boost::shared_ptr<StubValue>(new StubValue(finalStubNode));
             finalStub_->setID(finalStubIDRef_);
             IDManager::instance().setID(finalStubIDRef_,finalStub_);
          }
          else if(finalStubNode->Attribute("href")) { finalStubIDRef_ = finalStubNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { finalStub_ = boost::shared_ptr<StubValue>(new StubValue(finalStubNode));}
   }

    #ifdef ConsolePrint
        FileManager::instance().tap_ = initialtap_;
    #endif 
}
boost::shared_ptr<CalculationPeriodDatesReference> StubCalculationPeriodAmount::getCalculationPeriodDatesReference()
{
   if(!this->calculationPeriodDatesReferenceIsNull_){
        if(calculationPeriodDatesReferenceIDRef_ != ""){
             return boost::shared_static_cast<CalculationPeriodDatesReference>(IDManager::instance().getID(calculationPeriodDatesReferenceIDRef_));
        }else{
             return this->calculationPeriodDatesReference_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<CalculationPeriodDatesReference>();
   }
}
boost::shared_ptr<StubValue> StubCalculationPeriodAmount::getInitialStub()
{
   if(!this->initialStubIsNull_){
        if(initialStubIDRef_ != ""){
             return boost::shared_static_cast<StubValue>(IDManager::instance().getID(initialStubIDRef_));
        }else{
             return this->initialStub_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<StubValue>();
   }
}
boost::shared_ptr<StubValue> StubCalculationPeriodAmount::getFinalStub()
{
   if(!this->finalStubIsNull_){
        if(finalStubIDRef_ != ""){
             return boost::shared_static_cast<StubValue>(IDManager::instance().getID(finalStubIDRef_));
        }else{
             return this->finalStub_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<StubValue>();
   }
}
}

